WIP Enable configuration of TOP_DIR

gotcha opened the pull request:
used by `default_filter_filename`.

The test suite of my tool runs forked processes in a temporary directory (via `tempfile.mkdtemp`) which does not share the same root directory as the test suite.
This implies all calls are filtered out by `default_filter_filename`.

Adding the `set_top_dir` was enough to avoid filtering out.

This is a simple enough way to work around the little naive assumption made by:

    TOP_DIR = os.path.join(os.getcwd(), '')     # current dir with trailing slash

Would you see a better solution ?

gotcha committed:
Enable configuration of TOP_DIR

used by `default_filter_filename`.

gvanrossum reviewed (commented):
I would put the function directly after the definition of the variables, so it's harder for a reader to miss that these are not really constants.

gotcha committed:
move global TOP_DIR closer to where they are used.

gotcha commented:
I have moved the definition of the variables closer to where they are used.

See #112 to avoid global vars while still enabling top_dir customization.

gotcha committed:
Problem: 3.5.1 job fails

Solution: allow it to fail, following gvanrossum recommendation
to stop supporting 3.5

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/dropbox/pyannotate?pullRequest=111) <br/>All committers have signed the CLA.
